backend.go

```diff
@@ -60,6 +74,77 @@ func findAllBackends(mainLoose *fileStorer, mainPacked *pack.Storage, root strin
 	return storage, nil
 }
 
+func addAlternateDirectory(s []storage.Storage, dir string) ([]storage.Storage, error) {
+	s = append(s, newFileStorer(dir, ""))
+	pack, err := pack.NewStorage(dir)
+	if err != nil {
+		return s, err
+	}
+	s = append(s, pack)
+	return s, nil
+}
+
+func addAlternatesFromEnvironment(s []storage.Storage, env string) ([]storage.Storage, error) {
+	if len(env) == 0 {
+		return s, nil
+	}
+
+	for _, dir := range splitAlternateString(env, alternatesSeparator) {
+		var err error
+		s, err = addAlternateDirectory(s, dir)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return s, nil
+}
+
+var (
+	octalEscape = regexp.MustCompile("\\\\[0-7]{1,3}")
+	hexEscape   = regexp.MustCompile("\\\\x[0-9a-fA-F]{2}")
+)
+
+func splitAlternateString(env string, separator string) []string {
+	replacements := []struct {
```

Comment:
I'm nit-picking here, but I think that you could easily extract this out to the `var ( ... )` block above, which would make this method read a little bit more concisely.

Reply:
True, I can do that.